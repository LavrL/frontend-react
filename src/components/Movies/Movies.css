.Movies {
  text-align: center;
  position: relative;
  margin: 0;
}

.Movies::after {
  content: "";
  background-image: url('./img/avengers-movie.jpg');
  background-repeat: no-repeat repeat;
  background-size: cover;
  opacity: 0.4;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  position: fixed;
  z-index: -1;
}

.Movies-header {
  background-color: #282c34;
  height: 70px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  font-size: 20px;
  color: white;
  padding: 20px;
  cursor: pointer;
}

.spinner {
  height: 80px;
  margin: auto;
}

* {
  box-sizing: border-box;
}

.movies {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: center;
}

.Movies-header h2 {
  margin: 0;
}

.add-movies {
  text-align: center;
}

.add-movies button {
  font-size: 16px;
  padding: 8px;
  margin: 0 10px 30px 10px;
}

.button-currency__orange {
  height: 30px;
  font-size: 14px;
  border-radius: 8px;
  outline: none;
  border: 1px solid #fca311;
  background-color: #fca311;
  background-image: linear-gradient(-245deg, rgba(255, 255, 255, 0.3) 0%, rgba(255, 255, 255, 0) 100%);
  width: 150px;
}

.movie {
  padding: 5px 25px 30px 10px;
}

.errorMessage {
  margin: auto;
  font-weight: bold;
  color: rgb(161, 15, 15)
}

.search {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  margin-top: 10px;
}

input[type="submit"] {
  height: 30px;
  font-size: 14px;
  border-radius: 8px;
  outline: none;
  border: 1px solid #fca311;
  background-color: #fca311;
  background-image: linear-gradient(-245deg, rgba(255, 255, 255, 0.3) 0%, rgba(255, 255, 255, 0) 100%);
  width: 150px;
  margin-left: 5px;
}

input[type="submit"]:hover {
  background-color: #fca311;
  color: antiquewhite
}

.search>input[type="text"] {
  width: 40%;
  min-width: 170px;
}

.error-text {
  color: red;
  align-items: center;
  margin-top: 30px;
  font-size: 24px;
  font-weight: 600;
}

@media screen and (min-width: 694px) and (max-width: 915px) {
  .movie {
    max-width: 33%;
  }
}

@media screen and (min-width: 652px) and (max-width: 693px) {
  .movie {
    max-width: 50%;
  }
}

@media screen and (max-width: 651px) {
  .movie {
    max-width: 100%;
    margin: auto;
  }
}