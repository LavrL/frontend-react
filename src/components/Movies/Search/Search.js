import React, { useState } from "react";

const Search = ({ search }) => {
    const [searchValue, setSearchValue] = useState("");

    const handleSearchInputChanges = (e) => {
        setSearchValue(e.target.value);
    }

    const resetInputField = () => {
        setSearchValue("");
    }

    const callSearchFunction = (e) => {
        e.preventDefault();
        search(searchValue);
        resetInputField();
    }
    return (
        <div className="search">
            <input
                value={searchValue}
                onChange={handleSearchInputChanges}
                type="text"
                placeholder="Enter a movie: (ex. Batman)"
            />
            <input onClick={callSearchFunction}
                type="submit"
                value="SEARCH" />
        </div>
    )
}
export default Search;
