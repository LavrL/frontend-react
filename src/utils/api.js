export const API = 'https://api.openweathermap.org/data/2.5/weather';

export const api1 = (city, country, REACT_APP_WEATHER_APPID) => {
    return fetch(`${API}?q=${city},${country}&units=metric&appid=${REACT_APP_WEATHER_APPID}`)
        .then(res => { return res.json() });

};
