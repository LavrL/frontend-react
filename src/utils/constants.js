export const DEFAULT_IMAGE = 'https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_SX300.jpg';
export const MOVIE_API = "https://www.omdbapi.com/?s=batman&apikey=5d508a92";
export const REACT_APP_WEATHER_APPID = 'd416957097d97116bdbffca2895ada24'
