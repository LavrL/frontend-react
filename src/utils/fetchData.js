import axios from 'axios';

export const API = 'https://api.openweathermap.org/data/2.5/weather';
export const fetchData = async (city, country, REACT_APP_WEATHER_APPID) => {
    const url = `${API}?q=${city},${country}&units=metric&appid=${REACT_APP_WEATHER_APPID}`;

    return await axios.get(url);
};