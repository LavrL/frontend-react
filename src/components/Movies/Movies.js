import Header from "./Header/Header";
import Movie from "./Movie/Movie";
import React, { useEffect, useReducer } from 'react';
import Search from "./Search/Search";

import { MOVIE_API } from '../../utils/constants.js';
import './Movies.css';

const initialState = {
    loading: true,
    movies: [],
    errorMessages: null
}

export const reducer = (state = initialState, action) => {
    switch (action.type) {
        case "SEARCH_MOVIES_REQUEST":
            return {
                ...state,
                loading: true,
                errorMessages: null
            };
        case "SEARCH_MOVIES_SUCCESS":
            return {
                ...state,
                loading: false,
                movies: action.payload
            };
        case "SEARCH_MOVIES_FAILURE":
            return {
                ...state,
                loading: false,
                errorMessages: action.payload
            };
        default:
            return state;
    }
}

const Movies = () => {
    const [state, dispatch] = useReducer(reducer, initialState);

    useEffect(() => {
        fetch(MOVIE_API)
            .then(response => response.json())
            .then(jsonResponse => {
                dispatch({
                    type: "SEARCH_MOVIES_SUCCESS",
                    payload: jsonResponse.Search
                })
            })
            .catch(err => console.log(err))
    }, []);

    const search = searchValue => {
        dispatch({
            type: "SEARCH_MOVIES_REQUEST"
        });

        fetch(`https://www.omdbapi.com/?s=${searchValue}&apikey=5d508a92`)
            .then(response => response.json())
            .then(jsonResponse => {
                if (jsonResponse.Response === "True") {
                    dispatch({
                        type: "SEARCH_MOVIES_SUCCESS",
                        payload: jsonResponse.Search
                    })
                } else {
                    dispatch({
                        type: "SEARCH_MOVIES_FAILURE",
                        payload: jsonResponse.Error
                    })
                }
            })
            .catch(err => console.log(err));
    };

    const { movies, errorMessages, loading } = state;

    return (
        <div className="Movies">
            <Header text="MOVIE SEARCH" />
            <Search search={search} />
            <p className="Movies-intro">Find your favourite movies</p>
            <div className="movies">
                {loading && !errorMessages ? (<span>loading ... </span>) : (errorMessages !== null) ? (
                    <div className="error-text"> {errorMessages}</div>) : (
                        movies.map((movie, index) => (
                            <Movie key={`${index}-${movie.Title}`} movie={movie} />
                        ))
                    )}
            </div>
        </div>
    );
}

export default Movies;
