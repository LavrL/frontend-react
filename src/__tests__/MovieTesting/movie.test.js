import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Movies from '../../components/Movies/Movies';
import { reducer } from '../../components/Movies/Movies';
import Movie from '../../components/Movies/Movie/Movie';
import Search from '../../components/Movies/Search/Search';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import fetchMock from 'jest-fetch-mock';
import { DEFAULT_IMAGE } from '../../utils/constants';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
//import nock from 'nock';
//import fetch from 'isomorphic-fetch';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('axios');

configure({ adapter: new Adapter() });

describe('Movie testing', () => {

    let wrapper;
    let action;

    const initialState = {
        loading: true,
        movies: [],
        errorMessages: null
    };

    beforeAll(() => {
        wrapper = mount(<Movies />);
        fetchMock.resetMocks();
    });

    afterAll(() => {
        wrapper.unmount();
    });

    test('Movie component renders correctly', () => {
        const tree = renderer
            .create(<Movies />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('testing Movie component parameters', () => {
        const props = {
            movie: {
                Poster: 'Superman',
                Title: 'Superman returns',
                Year: 2015
            }
        };

        let movieComponent = mount(<Movie movie={props.movie} />);
        expect(movieComponent.find('h2').text()).toEqual('Superman returns');
    });

    test('testing default Poster Movie parameter', () => {
        const movie = {
            Poster: 'N/A'
        };
        let movieComponent = mount(<Movie movie={movie} />);
        expect(movieComponent.find('img').prop('src')).toEqual(DEFAULT_IMAGE);
    });

    test('reducer testing: SEARCH_MOVIES_REQUEST', () => {
        const mockStore = configureMockStore();
        const store = mockStore({});

        let state;
        expect(reducer(state, {})).toEqual(initialState)

        action = { type: "SEARCH_MOVIES_REQUEST" };
        expect(reducer(initialState, action))
            .toEqual({
                loading: true,
                errorMessages: null,
                movies: []
            });
    });

    test('reducer testing: SEARCH_MOVIES_SUCCESS', () => {
        action = { type: "SEARCH_MOVIES_SUCCESS" };
        expect(reducer(initialState, action))
            .toEqual({
                loading: false,
                errorMessages: null,
                movies: Array[
                    { 'Poster': "Superman", 'Title': "Superman returns" },
                    { 'Poster': "Batman", 'Title': "Batman returns" }
                ]
            });
    });

    test('reducer testing: SEARCH_MOVIES_FAILURE', () => {
        action = { type: "SEARCH_MOVIES_FAILURE" };
        expect(reducer(initialState, action))
            .toEqual({
                loading: false,
                errorMessages: undefined,
                movies: []
            });
    });

});