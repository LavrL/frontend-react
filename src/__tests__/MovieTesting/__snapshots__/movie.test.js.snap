// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie testing Movie component renders correctly 1`] = `
<div
  className="Movies"
>
  <header>
    <h2
      className="header__text"
    >
      MOVIE SEARCH
    </h2>
  </header>
  <div
    className="search"
  >
    <input
      onChange={[Function]}
      placeholder="Enter a movie: (ex. Batman)"
      type="text"
      value=""
    />
    <input
      onClick={[Function]}
      type="submit"
      value="SEARCH"
    />
  </div>
  <p
    className="Movies-intro"
  >
    Find your favourite movies
  </p>
  <div
    className="movies"
  >
    <span>
      loading ... 
    </span>
  </div>
</div>
`;
